app/main.go

```diff
@@ -62,8 +68,24 @@ func getIPAddress(r *http.Request) string {
 }
 
 func handlerPing(w http.ResponseWriter, r *http.Request) {
-	w.Write([]byte("pong"))
-	fmt.Println(time.Now(), getIPAddress(r), r.Method, r.RequestURI, r.UserAgent())
+	
+	requestBody, err := ioutil.ReadAll(r.Body)
+	n := rand.Intn(100)
+
+	<- time.After(time.Duration(lat) * time.Millisecond)
+	if n <= prob {
+		w.Write([]byte("pong"))
+		fmt.Println(time.Now(), getIPAddress(r), r.Method, r.RequestURI, r.UserAgent())
+	} else {
+		if err != nil {
+			w.WriteHeader(30)
+			log.Printf("%s: %v", string(requestBody), err.Error())
```

Comment:
Why we print request 3 times? (log.Printf, fmt.Printf, fmt.Println)

Reply:
I did not know which one to add so I just put them all :)

Reply:
=D